src/Voron/PageLocator.cs

```diff
@@ -1,15 +1,14 @@
+using System;
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
-using Sparrow.Server;
-using Voron.Impl;
 
 namespace Voron
 {
 
-    public sealed unsafe class PageLocator
+    public sealed class PageLocator
     {
-        [StructLayout(LayoutKind.Explicit, Size = 20)]
+        [StructLayout(LayoutKind.Explicit, Size = 32)]
```

Comment:
Why so big? You are wasting 7 bytes after `IsWritable`?

Reply:
To avoid accessing half cache. As we were already close to 32, it would make sense to just add 1 in more. But I can get away with 20 instead.

Reply:
Is it worth it to move to separate arrays? One for pages, one for the generation, one for writable (can bit bits field).
Less memory, more compact, and when you want to reset, you reset just the Generations array

Reply:
No, we are already paying for false sharing on a single array because we are not handling it in ways that CPU core affinity is respected. Paying for different arrays would also impact memory cache. It is faster to just fill everything with 0s every 128 iterations.